src/cdk/table/table.ts

```diff
@@ -357,11 +417,24 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
 
   /** Set up a subscription for the data provided by the data source. */
   private _observeRenderChanges() {
-    this._renderChangeSubscription = this.dataSource.connect(this).pipe(takeUntil(this._onDestroy))
-      .subscribe(data => {
-        this._data = data;
-        this._renderRowChanges();
-      });
+    let dataStream: Observable<T[]> | undefined;
+
+    if (this.dataSource instanceof DataSource) {
+      dataStream = this.dataSource.connect(this);
+    } else if (this.dataSource instanceof Observable) {
+      dataStream = this.dataSource;
+    } else if (this.dataSource instanceof Array) {
+      dataStream = of(this.dataSource);
+    }
+
+    if (dataStream === undefined) {
```

Comment:
What about cases where someone is binding to an array that hasn't been initialized yet?

Reply:
Ah good point, I think this would come up. I'll return if there is no data source.